### PR TITLE
Added additional logs to the tracking provider.

### PR DESCRIPTION
**Pull request by vinser52**
### Description
Added additional logs to the tracking provider.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: src/memory_pool.c [hunk 1]
```diff
@@ -59,6 +59,7 @@ static umf_result_t umfPoolCreateInternal(const umf_memory_pool_ops_t *ops,
     }
 
     *hPool = pool;
+    LOG_INFO("Memory pool created: %p", (void *)pool);
```
Comment (bratpiorka) [inline src/memory_pool.c:62]: I would like to keep one convention of prefixes.... sometimes you use a function name and sometimes you don't. Perhaps you could use "__FUNCTION__"?

File: src/provider/provider_tracking.c [hunk 2]
```diff
@@ -109,6 +118,9 @@ umf_result_t umfMemoryTrackerGetAllocInfo(const void *ptr,
     int found = critnib_find(TRACKER->map, (uintptr_t)ptr, FIND_LE,
                              (void *)&rkey, (void **)&rvalue);
     if (!found || (uintptr_t)ptr >= rkey + rvalue->size) {
+        LOG_WARN("umfMemoryTrackerGetAllocInfo: pointer %p not found in the "
+                 "tacker, TRACKER=%p",
```
Comment (kswiecicki) [inline src/provider/provider_tracking.c:122]: `tacker` -> `tracker`

File: src/provider/provider_tracking.c [hunk 3]
```diff
@@ -585,13 +597,20 @@ static umf_result_t trackingOpenIpcHandle(void *provider, void *providerIpcData,
 
     ret = umfMemoryProviderOpenIPCHandle(p->hUpstream, providerIpcData, ptr);
     if (ret != UMF_RESULT_SUCCESS) {
+        LOG_ERR(
+            "tracking open ipc handle: umfMemoryProviderOpenIPCHandle failed");
```
Comment (kswiecicki) [inline src/provider/provider_tracking.c:601]: I see that you start other log messages with a capital letter when its not a function name.

File: src/memory_pool.c (hunk 1 above)
Comment (vinser52) [reply src/memory_pool.c:62]: Done. I changed `LOG_*` macros and added `__func__` to it to automatically add function name to the log message.

File: src/provider/provider_tracking.c (hunk 2 above)
Comment (vinser52) [reply src/provider/provider_tracking.c]: Done

File: src/provider/provider_tracking.c (hunk 3 above)
Comment (vinser52) [reply src/provider/provider_tracking.c]: I refactored code: I changed LOG_* macros and added __func__ to it to automatically add function name to the log message.

**Review by bratpiorka — APPROVED** [review]

File: src/provider/provider_tracking.c [hunk 4]
```diff
@@ -256,33 +266,32 @@ static umf_result_t trackingAllocationMerge(void *hProvider, void *lowPtr,
     tracker_value_t *lowValue = (tracker_value_t *)critnib_get(
         provider->hTracker->map, (uintptr_t)lowPtr);
     if (!lowValue) {
-        LOG_ERR("tracking merge: no left value");
```
Comment (ldorau) [inline src/provider/provider_tracking.c:259]: Why do you remove this prefix?

Comment (ldorau) [reply src/provider/provider_tracking.c:259]: Ah, OK, you have added `__func__`

**Review by ldorau — APPROVED** [review]

File: src/provider/provider_tracking.c [hunk 5]
```diff
@@ -181,20 +190,21 @@ static umf_result_t trackingAllocationSplit(void *hProvider, void *ptr,
...
+                value->size, totalSize);
         ret = UMF_RESULT_ERROR_INVALID_ARGUMENT;
         goto err;
     }
 
     ret = umfMemoryProviderAllocationSplit(provider->hUpstream, ptr, totalSize,
                                            firstSize);
     if (ret != UMF_RESULT_SUCCESS) {
-        LOG_ERR("tracking split: umfMemoryProviderAllocationSplit failed");
+        LOG_ERR("umfMemoryProviderAllocationSplit failed");
```
Comment (kswiecicki) [inline src/provider/provider_tracking.c:207]: nit: In a few places we have log messages similar to `provider failed to split an allocation ...` and other times it's formulated like here `umfMemoryProviderAllocationSplit failed`.

Comment (vinser52) [reply src/provider/provider_tracking.c]: Done. I tried to use the first variant.

**Review by kswiecicki — APPROVED** [review]

**Pull request merged**